test/behavior/steps/login.py

```diff
@@ -0,0 +1,81 @@
+from behave import given, when, then
+
+
+keyword_dict = {inconsequential_input: 'something',
+				incorrect_input: 'incorrect'}
+
+
+@given('User is at the Tendrl login page')
+def step_impl(context):
+	context.browser.get(context.data['common']['tendrl']['ip'])
+
+
+@given('Tendrl API is not reachable')
+def step_impl(context):
+	pass
+
+
+@given('User is at the "cluster" list page')
+def step_impl(context):
+	context.browser.get(context.data['common']['tendrl']['ip'])
+	assert "cluster" in context.browser.current_url
+
+
+@when('User types in a correct "{field}"')
+def step_impl(context, field):
+	context.browser.find_element_by_id(field).send_keys(context.data['common']['tendrl'][field])
+
+
+@when('User types in an incorrect "{field}"')
+def step_impl(context, field):
+	context.browser.find_element_by_id(field).send_keys(keyword_dict[incorrect_input])
+
+
+@when('User types something in the "{field}" field')
+def step_impl(context, field):
+	context.browser.implicitly_wait(3)
+	context.browser.find_element_by_id(field).send_keys(keyword_dict[inconsequential_input])
+
+
+@when('User deletes what is in the "{field}" field')
+def step_impl(context, field):
+	context.browser.find_element_by_id(field).clear()
+
+
+@when('User clicks the Login button')
+def step_impl(context):
+	context.browser.find_element_by_class_name("login-btn").click()
+
+
+@when('User clicks on the Admin dropdown')
+def step_impl(context):
+	context.browser.find_element_by_id('usermenu').click()
+
+
+@when('User clicks on the Logout button')
+def step_impl(context):
+	context.browser.find_element_by_id('logoutButton').click()
+
+
+@then('The text "{field} is required" should appear under the "{field}" box')
+def step_impl(context, field):
+	element_id = field + "HelpBlock"
+	block_text = context.browser.find_element_by_id(element_id)
+	assert block_text == field + " is required"
+
+
+@then('The {message} popup should appear above the {field} box')
+def step_impl(context, message, field):
+	message = message[1:-1]
```

Comment:
why are you doing this?

Reply:
Issue regarding quotation marks; for some reason they are included as part of the string in this particular case and need to be chopped off

Reply:
can you search for the solution for this? I think this is not a good way of handling it.

Or we can get on a call on friday and discuss it.

Reply:
I'll look into it and see if I can find another solution